### PR TITLE
Break current culture initialization cycle

### LINKED ISSUE #15931: [Arm64/Linux] New regression on tip
Several tests are failing with
```
FailFast: Infinite recursion during resource lookup within System.Private.CoreLib.  
This may be a bug in System.Private.CoreLib, or potentially in certain extensibility points
such as assembly resolve events or CultureInfo names.  
Resource name: Arg_NullReferenceException
```
List of affected tests
```
CoreMangLib/cti/system/double/DoubleToString2/DoubleToString2.sh
Interop/MarshalAPI/String/StringMarshalingTest/StringMarshalingTest.sh
JIT/Regression/CLR-x86-JIT/V1-M09/b15307/b15307/b15307.sh
JIT/Regression/CLR-x86-JIT/V1-M10/b04914/b04914/b04914.sh
Regressions/common/ThreadCulture/ThreadCulture.sh
Regressions/coreclr/0202/ThreadCulture/ThreadCulture.sh
Regressions/coreclr/0570/Test570/Test570.sh
Regressions/coreclr/0576/Test0576/Test0576.sh
Regressions/coreclr/1402/Test1402/Test1402.sh
```
`git bisect bad f1c732878`
`git bisect good 84efbc8`

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #15931

### CODE REVIEW

**Comment by jkotas** [comment]
@dotnet-bot help

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by tarekgh** [comment]
do we understand why the problem in #15931 start to show up recently?

**Comment by jkotas** [comment]
This is fixing regression introduced by #15902.

**Comment by jkotas** [comment]
test Ubuntu x64 Checked corefx_baseline please
test Windows_NT x64 Checked corefx_baseline please

**Comment by tarekgh** [comment]
test Ubuntu x64 Checked Innerloop Build and Test

File: src/mscorlib/src/System/Globalization/CultureInfo.cs [hunk 1]
```diff
@@ -920,7 +893,9 @@ public virtual DateTimeFormatInfo DateTimeFormat
 
         public void ClearCachedData()
         {
-            Init(); // reset the default culture values
+            // reset the default culture values
+            s_userDefaultCulture = GetUserDefaultCulture();
```
Comment (drieseng) [inline src/mscorlib/src/System/Globalization/CultureInfo.cs]: Wouldn't it be better to just reset these to null?
That way, these will again be lazily initialized if **UserDefaultUICulture** or **InstalledUICulture** are used again.
If however they are not used again (before the next invocation of `ClearCachedData()`), you're saving some cycles.

Note that you'd have to modify `InitializeUserDefaultCulture()` and `InitializeUserDefaultUICulture()` to take this into account.

Comment (jkotas) [reply src/mscorlib/src/System/Globalization/CultureInfo.cs]: ClearCachedData method is very rarely used. I think that the current simple and less efficient implementation is the right trade-off.

Thanks for the feedback anyway.

**Pull request merged**